samza-core/src/main/scala/org/apache/samza/config/MetricsConfig.scala

```diff
@@ -44,9 +47,36 @@ class MetricsConfig(config: Config) extends ScalaMapConfig(config) {
   def getMetricsReporterInterval(name: String): Option[String] = getOption(MetricsConfig.METRICS_SNAPSHOT_REPORTER_INTERVAL format name)
 
   /**
-   * Returns a list of all metrics names from the config file. Useful for
-   * getting individual metrics.
-   */
+    * Parses the metric key name specified in config.
+    * Customers can specify
+    * hostname (for partitioning by hostname -- default) or jobname or jobid
+    *
+    * @param name The name of the reporter
+    * @return the metric key name (if specified), else None
+    */
+  def getMetricsReporterStreamPartitionKeyName(name: String): Option[String] = getOption(MetricsConfig.METRICS_SNAPSHOT_REPORTER_PARTITION_KEY format name)
+
+  /**
+    * Returns the actual value of the partition key, based on the keyName.
+    * If specified as "jobname" returns the jobname.
+    * If none is specified or is specified as "hostname", returns the hostname,
+    * else returns the actual value specified.
+    *
+    * @param keyName  the desired key name (hostname, jobname, none, or another custom name)
+    * @param jobName  the current job name
+    * @param hostname the current host name
+    */
+  def getMetricsReporterStreamPartitionKeyValue(keyName: Option[String], jobName: String, hostname: String) = keyName match {
```

Comment:
prefer moving this method out of the MetricReporterConfig and instead adding it to `SnapshotReporter` instead.